### PR TITLE
10 get api articles

### PR DESCRIPTION
**Pull request by cutetofuu**
Do I need to test if the `sort_by` query works for all columns, or is testing for columns such as `article_id` and `title` sufficient?

Would `body` and `article_img_url `count as valid sort by options? All of the other columns make sense, but it seems a bit strange to sort the articles based on either of these two columns.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -103,6 +103,94 @@ describe("articles", () => {
...
+    });
+    it("404: non-existent topic given", () => {
+      return request(app)
+        .get("/api/articles?topic=1000")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("No article found");
+        });
+    });
+    it("200: sorts articles by title in descending order", () => {
```
Comment (JVal-NC) [inline __tests__/app.test.js:138]: Have a look at jest-sorted npm package

File: __tests__/app.test.js [hunk 2]
```diff
@@ -103,6 +103,94 @@ describe("articles", () => {
     });
   });
 
+  describe("GET /api/articles (queries)", () => {
```
Comment (JVal-NC) [inline __tests__/app.test.js:106]: Good tests, how about a valid query with no data within it?

File: models/articles.models.js [hunk 3]
```diff
@@ -1,27 +1,59 @@
...
-              COUNT(comments.article_id)::INTEGER AS comment_count
-          FROM articles
-          LEFT JOIN comments ON articles.article_id = comments.article_id
-          GROUP BY articles.article_id
-          ORDER BY articles.created_at DESC
-      `
-    )
-    .then(({ rows }) => {
+exports.fetchArticles = (topic, sort_by, order) => {
+  const validSortByOptions = [
```
Comment (JVal-NC) [inline models/articles.models.js:4]: Good valid sortby array, could we do similar with orderby?

**Review by JVal-NC — COMMENTED** [review]
Good work, couple comments in the attached PR.

**Review by jimstev2447 — APPROVED** [review]

**Pull request merged**